src/test_class.rs

```diff
@@ -112,7 +114,7 @@ impl<B: UsbBus> TestClass<'_, B> {
         &self,
         usb_bus: &'a UsbBusAllocator<B>,
     ) -> UsbDeviceBuilder<'a, B> {
-        UsbDeviceBuilder::new(usb_bus, UsbVidPid(VID, PID))
+        UsbDeviceBuilder::new(usb_bus, UsbVidPid(VID, PID), unsafe { &mut CONTROL_BUFFER })
```

Comment:
Not an issue really, but I get the following warning with 1.77.0-nightly
```
warning: mutable reference of mutable static is discouraged
   --> .../usb-device/src/test_class.rs:117:70
    |
117 |         UsbDeviceBuilder::new(usb_bus, UsbVidPid(VID, PID), unsafe { &mut CONTROL_BUFFER })
    |                                                                      ^^^^^^^^^^^^^^^^^^^ mutable reference of mutable static
    |
    = note: for more information, see issue #114447 <https://github.com/rust-lang/rust/issues/114447>
    = note: reference of mutable static is a hard error from 2024 edition
    = note: mutable statics can be written to by multiple threads: aliasing violations or data races will cause undefined behavior
    = note: `#[warn(static_mut_ref)]` on by default
help: mutable references are dangerous since if there's any other pointer or reference used for that static while the reference lives, that's UB; use `addr_of_mut!` instead to create a raw pointer
    |
117 |         UsbDeviceBuilder::new(usb_bus, UsbVidPid(VID, PID), unsafe { addr_of_mut!(CONTROL_BUFFER) })
    |                                                                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
Hmm, this is indeed a strange problem to solve for embedded. Using the mentioned macro would force us to change the API to accept a raw pointer instead of a slice reference, which really tanks usability for a non-`static mut` buffer. Let me take a look at this and figure out the right approach.

Reply:
Based on the comments on the referenced issue, the approach was to use an `UnsafeCell` around the buffer, which is very doable. I have updated the implementation and docs. Thanks for pointing this out :)